core/src/transport.rs

```diff
@@ -57,6 +57,17 @@ pub use self::optional::OptionalTransport;
 pub use self::upgrade::Upgrade;
 
 static NEXT_LISTENER_ID: AtomicUsize = AtomicUsize::new(1);
+#[derive(Debug, Default, Copy, Clone)]
+pub enum PortMode {
+    #[default]
+    New,
+    Reuse,
+}
```

Comment:
Nit: We can bikeshed this at a later point but I am not really a fan of the "Mode" wording. Perhaps something like `PortAllocation` would be better. Or `PortUse`?

Reply:
PortUse sounds good to me